### DISCUSSION TITLE
How do I notice content overflow for slides in RevealJS?

### DISCUSSION THREAD

**Discussion by kcarnold** [quarto-dev/quarto-cli#6824] [Q&A]
### Description

I have made various revealjs presentations and am struggling with fitting everything on a slide. (e.g., plot bottoms getting cut off; I'm using plotly)

There are definitely individual things I need to learn about, like how to tweak Plotly heights. And I may be tripping on some actual layout bugs; I can follow up later with those. But for this question I just want to ask: **how do you systematically notice when you have a slide with overflow**?

Usually I'm working with the preview window open, so I see the effect right after making it. But sometimes I miss things, especially when there's a fragment or I'm working on a different part of the slide and didn't notice it pushed something else down. Then I notice during the presentation, and it's usually too late to fix (though I just started turning on `scrollable: true` to work around that).

Short of carefully looking through everything, is there any way to get a warning about overflowing elements on slides?

**Comment by mcanouil** [comment]
You could write a JavaScript script that check client height versus scroll/element height maybe.
Note that this is more of an HTML question.

Meanwhile, you should always go through your slides before a presentation and possibly use the "print to PDF" feature which could also help, especially with overflowing images (see <https://quarto.org/docs/presentations/revealjs/presenting.html#print-to-pdf>).
